File: api/app/models/report.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, text
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Report(Base):
    __tablename__ = "reports"

    id: Mapped[uuid.UUID] = mapped_column(
        primary_key=True, server_default=text("gen_random_uuid()")
    )
    inspection_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("inspections.id", ondelete="CASCADE"), nullable=False
    )
    status: Mapped[str] = mapped_column(
        String(20), nullable=False, default="pending"
    )  # pending / in_progress / resolved
    action_checks: Mapped[dict | None] = mapped_column(
        JSONB, nullable=True
    )  # [false, true, false] — guidance.action_steps 인덱스 대응
    resolver_id: Mapped[uuid.UUID | None] = mapped_column(
        ForeignKey("users.id"), nullable=True
    )
    resolved_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    note: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=text("now()")
    )

File: app/lib/models/report.dart
class Report {
  const Report({
    required this.id,
    required this.inspectionId,
    required this.status,
    required this.actionChecks,
    required this.createdAt,
    this.resolverId,
    this.resolvedAt,
    this.note,
  });

  final String id;
  final String inspectionId;
  final String status;
  final List<bool> actionChecks;
  final String? resolverId;
  final DateTime? resolvedAt;
  final String? note;
  final DateTime createdAt;

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'] as String,
      inspectionId: json['inspection_id'] as String,
      status: json['status'] as String,
      actionChecks: (json['action_checks'] as List<dynamic>?)
              ?.map((e) => e as bool)
              .toList() ??
          [],
      resolverId: json['resolver_id'] as String?,
      resolvedAt: json['resolved_at'] != null
          ? DateTime.parse(json['resolved_at'] as String)
          : null,
      note: json['note'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'inspection_id': inspectionId,
      'status': status,
      'action_checks': actionChecks,
      'note': note,
    };
  }
}
